modeshape-jcr/src/main/java/org/modeshape/jcr/AccessControlManagerImpl.java

```diff
@@ -261,9 +262,9 @@ private JcrAccessControlList findAccessList( String absPath,
         SessionCache sessionCache = session.cache();
         Map<String, Set<String>> permissions = startingNode.getPermissions(sessionCache);
         CachedNode node = startingNode;
-        if (permissions == null && searchParents) {
-            // walk up the hierarchy until we get a set of permissions or we reach the root or a missing parent
-            while (permissions == null) {
+        // walk up the hierarchy until we get a set of non-empty permissions or we reach the root or a missing parent
+        if ((permissions == null || permissions.isEmpty()) && searchParents) {
```

Comment:
Minor nit: couldn't this line just be `if ( searchParents ) {` since the `permission`-related criteria would be handled in the `while` loop?